lightning/src/ln/onion_utils.rs

```diff
@@ -935,6 +935,29 @@ pub(crate) fn decode_next_payment_hop<NS: Deref>(
 	}
 }
 
+/// Build a payment onion, returning the first hop msat and cltv values as well.
+pub fn create_payment_onion<T>(
+	secp_ctx: &Secp256k1<T>, path: &Path, session_priv: &SecretKey, total_msat: u64, 
+	recipient_onion: RecipientOnionFields, best_block_height: u32, payment_hash: PaymentHash,
+	keysend_preimage: Option<PaymentPreimage>, prng_seed: [u8; 32]
+) -> Result<(u64, u32, msgs::OnionPacket), ()>
+where
+	T: secp256k1::Signing
+{
+	let onion_keys = construct_onion_keys(&secp_ctx, &path, &session_priv).map_err(|_| ())?;
+	let (onion_payloads, htlc_msat, htlc_cltv) = build_onion_payloads(
+		&path,
+		total_msat,
+		recipient_onion,
+		best_block_height + 1,
+		&keysend_preimage,
```

Comment:
nit: we don't do the rustfmt vertical thing

```suggestion
		&path, total_msat, recipient_onion, best_block_height + 1, &keysend_preimage
```